ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -300,23 +320,30 @@ private SourceDescription describe(String name, boolean extended) throws KsqlExc
 
     StructuredDataSource dataSource = ksqlEngine.getMetaStore().getSource(name);
     if (dataSource == null) {
-      throw new KsqlException(String.format("Could not find STREAM/TABLE '%s' in the Metastore", name));
+
+      PersistentQueryMetadata metadata = ksqlEngine.getPersistentQueries().get(new QueryId(name));
+      // check if we are describing a Query (not found in the metastore)
+      if (metadata == null) {
+        throw new KsqlException(String.format("Could not find STREAM/TABLE/QUERY '%s' in the Metastore", name));
+      }
+      KsqlStructuredDataOutputNode outputNode = (KsqlStructuredDataOutputNode) metadata.getOutputNode();
+      return new SourceDescription(outputNode, metadata.getStatementString(), name, metadata.getTopoplogy(), metadata.getExecutionPlan(), true, ksqlEngine.getTopicClient());
```

Comment:
It looks like queries will always show the extended output. The seventh argument is `true` instead of `extended`. Is my understanding correct? If so, is this intentional?

If not intentional, we should add a test for this.

Reply:
Thanks - I have refactored this out - using EXPLAIN QUERY-ID instead DESCRIBE